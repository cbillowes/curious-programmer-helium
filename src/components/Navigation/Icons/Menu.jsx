import React, { Component } from "react"
import "./Menu.css"

class Menu extends Component {
  render() {
    return (
      <div
        className={`icon menu ${this.props.active ? "active" : ""}`}
        onClick={this.props.toggle.bind(this)}
      >
        <i />
      </div>
    )
  }
}

export default Menu