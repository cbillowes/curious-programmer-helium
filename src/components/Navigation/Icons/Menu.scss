.navigation .icons .icon.menu i {
    background-image: url(data:image/svg+xml;utf8;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iaXNvLTg4NTktMSI/Pgo8IS0tIEdlbmVyYXRvcjogQWRvYmUgSWxsdXN0cmF0b3IgMTkuMC4wLCBTVkcgRXhwb3J0IFBsdWctSW4gLiBTVkcgVmVyc2lvbjogNi4wMCBCdWlsZCAwKSAgLS0+CjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayIgdmVyc2lvbj0iMS4xIiBpZD0iQ2FwYV8xIiB4PSIwcHgiIHk9IjBweCIgdmlld0JveD0iMCAwIDUxMiA1MTIiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDUxMiA1MTI7IiB4bWw6c3BhY2U9InByZXNlcnZlIiB3aWR0aD0iNTEycHgiIGhlaWdodD0iNTEycHgiPgo8Zz4KCTxnPgoJCTxnPgoJCQk8Y2lyY2xlIGN4PSIyNTYiIGN5PSIyNTYiIHI9IjY0IiBmaWxsPSIjRkZGRkZGIi8+CgkJCTxjaXJjbGUgY3g9IjI1NiIgY3k9IjQ0OCIgcj0iNjQiIGZpbGw9IiNGRkZGRkYiLz4KCQkJPGNpcmNsZSBjeD0iMjU2IiBjeT0iNjQiIHI9IjY0IiBmaWxsPSIjRkZGRkZGIi8+CgkJPC9nPgoJPC9nPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+CjxnPgo8L2c+Cjwvc3ZnPgo=);
}
