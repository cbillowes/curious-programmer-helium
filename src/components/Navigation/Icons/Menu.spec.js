import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import Menu from "./Menu"

describe("a menu icon component", () => {
  let element = <Menu toggle={() => console.log("toggle")} active="true" />
  let component
  
  beforeAll(() => {
    component = mount(element)    
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })
})