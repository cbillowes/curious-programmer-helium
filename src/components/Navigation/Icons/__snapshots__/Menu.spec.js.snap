// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a menu icon component should render 1`] = `
<div
  className="icon menu active"
  onClick={[Function]}
>
  <i />
</div>
`;
